### DISCUSSION TITLE
Keeping NR.dmnd in the RAM

### DISCUSSION THREAD

**Discussion by blaizereal** [bbuchfink/diamond#552] [Ideas]
Would an option to keep the database in RAM make sense? At this time it takes about 150-170sec to load the entire NR.dmnd (220GB) into memory. (Dual Epyc, PCIe v3 - 8GT/s, NVRAM, 384Gb/2666MHz RAM based server). 
For a few samples, this obviously doesn't cause much delay, but for e.g. 300 samples, it takes ~13 hours of extra computing time to load the whole database over and over again. 

Thanks: B

**Comment by bbuchfink** [comment]
You can use BLAST databases for that purpose which support memory mapped files, that should make loading substantially faster. Keeping the database in RAM may make sense if you want to run a lot of small query files, but the best option would be to merge them into a larger query batch. I will still consider this as a possible feature when I find the time.

**Reply by blaizereal** [reply]
Thanks for the idea, I will definitely try it!

B
